book/src/asm/functions.md

```diff
@@ -24,13 +24,13 @@ Labels allow referring to a location in a function by name.
 
 ### Assignments
 
-Assignments allow setting the value of a write register to the value of an [expression](#expressions) using an assignment register.
+Assignments allow setting the values of some write registers to the values of some expressions [expression](#expressions) using assignment registers.
 
 ```
 {{#include ../../../test_data/asm/book/function.asm:instruction}}
 ```
 
-One important requirement is for the assignment register of the assignment to be compatible with that of the expression. This is especially relevant for instructions: the assignment register of the instruction output must match that of the assignment. In this example, we use `Y` in the assignment as the output of `square` is `Y`:
+One important requirement is for the assignment registers of the assignment to be compatible with those of the expressions. This is especially relevant for instructions: the assignment registers of the instruction output must match that of the assignment. In this example, we use `Y` in the assignment as the output of `square` is `Y`:
```

Comment:
Changed the paragraph a bit

Reply:
That's the annoying thing with multi return. I do think that it's easier to model as the RHS being many expressions, simply you can only construct the multi return case with a function call. The reason is that if it's a single expression, it should have a type, but here it would have a list.

Reply:
tuples should be a type :grimacing:

Reply:
Yes, once we have types, it should be a single tuple expression. Now it is more similar to go multi returns.